qualtran/bloqs/factoring/ecc/ec_phase_estimate_r.py

```diff
@@ -48,27 +50,54 @@ class ECPhaseEstimateR(Bloq):
     Args:
         n: The bitsize of the elliptic curve points' x and y registers.
         point: The elliptic curve point to phase estimate against.
+        ec_window_size: The number of bits in the ECAdd window.
+        mul_window_size: The number of bits in the modular multiplication window.
     """
 
     n: int
     point: ECPoint
+    ec_window_size: int = 1
+    mul_window_size: int = 1
 
     @cached_property
     def signature(self) -> 'Signature':
         return Signature([Register('x', QUInt(self.n)), Register('y', QUInt(self.n))])
 
+    @property
+    def ec_add(self) -> Union[functools.partial[ECAddR], functools.partial[ECWindowAddR]]:
+        if self.ec_window_size == 1:
+            return functools.partial(ECAddR, n=self.n)
+        return functools.partial(
+            ECWindowAddR,
+            n=self.n,
+            ec_window_size=self.ec_window_size,
+            mul_window_size=self.mul_window_size,
+        )
+
     def build_composite_bloq(self, bb: 'BloqBuilder', x: Soquet, y: Soquet) -> Dict[str, 'SoquetT']:
         if isinstance(self.n, sympy.Expr):
             raise DecomposeTypeError("Cannot decompose symbolic `n`.")
         ctrl = [bb.add(PlusState()) for _ in range(self.n)]
-        for i in range(self.n):
-            ctrl[i], x, y = bb.add(ECAddR(n=self.n, R=2**i * self.point), ctrl=ctrl[i], x=x, y=y)
+
+        if self.ec_window_size == 1:
+            for i in range(self.n):
+                ctrl[i], x, y = bb.add(self.ec_add(R=2**i * self.point), ctrl=ctrl[i], x=x, y=y)
+        else:
+            ctrls = np.split(np.array(ctrl), self.n // self.ec_window_size)
+            for i in range(self.n // self.ec_window_size):
```

Comment:
optional: you can probably factor out `self.n // self.ec_window_size` as a quick lil property on this bloq